.github/workflows/build.yaml

```diff
@@ -327,7 +335,7 @@ jobs:
               run: |
                   for BUILD_TYPE  in clang python_lib; do
                       case $BUILD_TYPE in
-                          "clang") GN_ARGS='is_clang=true target_os="all" is_asan=true pw_command_launcher="`pwd`/../scripts/helpers/clang-tidy-launcher.py"';;
```

Comment:
Not running the bugprone tidy bits on Darwin is not OK.

Reply:
Updated to run clang-tidy. Will double-check with CI that relevant files are covered (scripts are a bit harder to track).

Reply:
Updated to run clang and found some issues:
  - it seems we attempted to compile ios using `target_os="all"` (or I believe we did) however because iOS is always build with gcc, this only made mac builds with all clang and gcc combinations 

I updated to compile mac and ios in separate steps and only used the mac for tidy since this is what runs under clang.

Reply:
It all runs under clang. On darwin gcc is a symlink to clang.

Reply:
How should I handle this? From the way I read this, the intent of this command is to have 'target_os = all' which seems to imply more builds (at least I would expect ios) and it says 'clang=true' to force clang.

Instead it builds `mac_clang` , `mac_gcc` (which is duplicated with clang from what you say) and `mac_gcc_mbedtls` and does not build any ios.

Should we force ios toolchains to be clang rather than gcc?

Reply:
I am not sure what the story here is.  @vivien-apple or @mspang might know....

Reply:
This should be fixed after #16551 

clang-tidy now runs over the same dataset. I also updated the script to consider both clang and gcc invocations for tidy.